conans/client/graph/install_graph.py

```diff
@@ -454,47 +454,69 @@ def install_build_order(self):
                   "order": [[n.serialize() for n in level] for level in install_order]}
         return result
 
-    def raise_errors(self):
+    def get_missing_invalid_packages(self):
+        def analyze_package(package):
+            if package.binary == BINARY_MISSING:
+                missing.append(package)
+            elif package.binary == BINARY_INVALID:
+                invalid.append(package)
         missing, invalid = [], []
-        for ref, install_node in self._nodes.items():
-            for package in install_node.packages.values():
-                if package.binary == BINARY_MISSING:
-                    missing.append(package)
-                elif package.binary == BINARY_INVALID:
-                    invalid.append(package)
+        for _, install_node in self._nodes.items():
+            if self._order == "recipe":
+                for package in install_node.packages.values():
+                    analyze_package(package)
+            elif self._order == "configuration":
+                analyze_package(install_node)
 
+        return missing, invalid
+
+    def raise_errors(self, basic_output=False):
+        missing, invalid = self.get_missing_invalid_packages()
         if invalid:
-            msg = ["There are invalid packages:"]
-            for package in invalid:
-                node = package.nodes[0]
-                if node.cant_build and node.should_build:
-                    binary, reason = "Cannot build for this configuration", node.cant_build
-                else:
-                    binary, reason = "Invalid", node.conanfile.info.invalid
-                msg.append("{}: {}: {}".format(node.conanfile, binary, reason))
-            raise ConanInvalidConfiguration("\n".join(msg))
+            self._raise_invalid(invalid)
         if missing:
-            self._raise_missing(missing)
+            self._raise_missing(missing, basic_output)
+
+    def get_errors(self):
+        try:
+            self.raise_errors(basic_output=True)
+        except ConanException as e:
+            return str(e)
+        return None
```

Comment:
It looks like that instead of modifying the ``_raise_missing`` and doing this, it would be better to craft its own simpler message here.

Reply:
But `raise_errors` could potentially raise invalid binaries errors also.

Reply:
Yes, sure, it can iterate all packages and add an error message line for each invalid and missing.